torch/nn/_functions/rnn.py

```diff
@@ -43,17 +46,18 @@ def LSTMCell(input, hidden, w_ih, w_hh, b_ih=None, b_hh=None):
 
 
 def GRUCell(input, hidden, w_ih, w_hh, b_ih=None, b_hh=None):
-
     if input.is_cuda:
-        gi = F.linear(input, w_ih)
+        gi = input.expand(3, input.size(0), input.size(1)).transpose(0, 1).contiguous() if w_ih is None else \
+            F.linear(input, w_ih)
         gh = F.linear(hidden, w_hh)
         state = fusedBackend.GRUFused()
         return state(gi, gh, hidden) if b_ih is None else state(gi, gh, hidden, b_ih, b_hh)
 
-    gi = F.linear(input, w_ih, b_ih)
+    gi = input.expand(3, input.size(0), input.size(1)).transpose(0, 1).contiguous() if w_ih is None else \
+        F.linear(input, w_ih, b_ih)
     gh = F.linear(hidden, w_hh, b_hh)
-    i_r, i_i, i_n = gi.chunk(3, 1)
-    h_r, h_i, h_n = gh.chunk(3, 1)
+    i_r, i_i, i_n = torch.unbind(gi.view(input.size(0), 3, -1), 1)
+    h_r, h_i, h_n = torch.unbind(gh.view(input.size(0), 3, -1), 1)
```

Comment:
Was having difficulties with getting chunk working well since it keeps an extra dimension. I could try to work around this but it ended up in more headache, however if there is a compelling reason to use chunk rather than unbind I can try to figure a solution out!